guide/getting-started-nodejs.md

```diff
@@ -0,0 +1,139 @@
+## Getting started with NodeJS on OpenFaaS
+
+There are a number of resources that document getting started with
+OpenFaaS, in fact the entire stack can be deployed in less than 60
+seconds when following the instructions from the README on GitHub!
+This article will focus on how easy it is to setup a simple NodeJS
+function and deploy it.
+
+#### Dependencies
+
+This article assumes installation of at least `Docker 17.05`. Docker
+provides the bare minimum necessary to build and deploy function but
+there is a better tool for the job!
+
+###### faas-cli
+
+The `faas-cli` tool is a complementary package that significantly simplifies the
+function build and deployment process. To install:
+```
+curl -sSL https://cli.openfaas.com | sudo sh
+```
+
+...and that's it! Docker and faas-cli are the only two
+dependencies needed to complete the steps below.
+
+#### Initializing the OpenFaaS Service
+
+We have our dependencies, now let's launch OpenFaaS so we have
+somewhere to deploy our function.
+
+Grab the code:
+```
+git clone https://github.com/alexellis/faas.git && cd faas
+```
+
+...initialize Docker Swarm
+```
+docker swarm init
+```
+
+...and deploy OpenFaaS on Docker Swarm
+```
+./deploy_stack.sh
+```
+
+#### Building our function
+
+Let's build out a simple function directory...
+```
+mkdir -p ./functions/hello-node && cd ./functions
+```
+
+...and create a handler.js file that will house our function.
+```
+echo "module.exports = (req) => console.log('Hello! You said:', req)" > ./hello-node/handler.js
```

Comment:
The echo trick is cool, but maybe we could write the text completely for readability?

Here is an example:

https://docs.docker.com/engine/userguide/eng-image/multistage-build/#before-multi-stage-builds